test/framework/src/main/java/org/opensearch/test/OpenSearchTestClusterRule.java

```diff
@@ -0,0 +1,397 @@
+                if (afterClass || currentClusterScope == Scope.TEST) {
+                    cluster().close();
+                }
+                cluster().assertAfterTest();
+            }
+        } finally {
+            if (currentClusterScope == Scope.TEST) {
+                clearClusters(); // it is ok to leave persistent / transient cluster state behind if scope is TEST
+            }
+        }
+    }
+
+    private void clearClusters() throws Exception {
+        synchronized (clusters) {
+            if (!clusters.isEmpty()) {
+                IOUtils.close(clusters.values());
+                clusters.clear();
+            }
+        }
+        if (restClient != null) {
+            restClient.close();
+            restClient = null;
+        }
+        OpenSearchTestCase.assertBusy(() -> {
+            int numChannels = RestCancellableNodeClient.getNumChannels();
+            OpenSearchTestCase.assertEquals(
+                numChannels
+                    + " channels still being tracked in "
+                    + RestCancellableNodeClient.class.getSimpleName()
+                    + " while there should be none",
+                0,
+                numChannels
+            );
+        });
+    }
+
+    private Statement statement(final Statement base, FrameworkMethod method, Object target) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                before(target, method);
+
+                List<Throwable> errors = new ArrayList<Throwable>();
+                try {
+                    base.evaluate();
+                } catch (Throwable t) {
+                    errors.add(t);
+                } finally {
+                    try {
+                        after(target, method);
+                    } catch (Throwable t) {
+                        errors.add(t);
+                    }
+                }
+                MultipleFailureException.assertEmpty(errors);
+            }
+        };
+    }
+
+    private void initializeSuiteScope(OpenSearchIntegTestCase target, FrameworkMethod method) throws Exception {
+        final Class<?> targetClass = getTestClass();
+        /*
+          Note we create these test class instance via reflection
+          since JUnit creates a new instance per test.
+         */
+        if (suiteInstance != null) {
+            // Catching the case when parameterized test cases are run: the test class stays the same but the test instances changes.
+            if (target instanceof ParameterizedOpenSearchIntegTestCase) {
+                assert suiteInstance instanceof ParameterizedOpenSearchIntegTestCase;
+                if (hasParametersChanged((ParameterizedOpenSearchIntegTestCase) target)) {
+                    printTestMessage("new instance of parameterized test class, recreating cluster scope", method);
+                    afterClass();
+                    beforeClass();
```

Comment:
Does this logic in effect change a parameterized SUITE scope test into to a TEST scope test? I believe parameterized tests generally run one test method with all parameterized values, then the next test method with all parameterized values, which means every test invocation is going to see a change in parameters here.

Reply:
> Does this logic in effect change a parameterized SUITE scope test into to a TEST scope test? 

No, this logic kicks off only when all tests from the sites are executed and new parameter(s) combination is injected so the suite needs to be executed again (a bit more elaborate example of how it works https://github.com/opensearch-project/OpenSearch/pull/11877#issuecomment-1896570247

Reply:
Ah, I was wrong about how the parameterized testing framework orchestrated the tests. Thanks!